Index: llvm-toolchain-snapshot_15~++20220407011631+46f0e2ceb487/openmp/runtime/src/kmp_affinity.h
===================================================================
--- llvm-toolchain-snapshot_15~++20220407011631+46f0e2ceb487.orig/openmp/runtime/src/kmp_affinity.h
+++ llvm-toolchain-snapshot_15~++20220407011631+46f0e2ceb487/openmp/runtime/src/kmp_affinity.h
@@ -232,28 +232,29 @@ public:
 #elif __NR_sched_getaffinity != 223
 #error Wrong code for getaffinity system call.
 #endif /* __NR_sched_getaffinity */
-#elif KMP_ARCH_MIPS
-#ifndef __NR_sched_setaffinity
-#define __NR_sched_setaffinity 4239
-#elif __NR_sched_setaffinity != 4239
-#error Wrong code for setaffinity system call.
-#endif /* __NR_sched_setaffinity */
-#ifndef __NR_sched_getaffinity
-#define __NR_sched_getaffinity 4240
-#elif __NR_sched_getaffinity != 4240
-#error Wrong code for getaffinity system call.
-#endif /* __NR_sched_getaffinity */
-#elif KMP_ARCH_MIPS64
-#ifndef __NR_sched_setaffinity
-#define __NR_sched_setaffinity 5195
-#elif __NR_sched_setaffinity != 5195
-#error Wrong code for setaffinity system call.
-#endif /* __NR_sched_setaffinity */
-#ifndef __NR_sched_getaffinity
-#define __NR_sched_getaffinity 5196
-#elif __NR_sched_getaffinity != 5196
-#error Wrong code for getaffinity system call.
-#endif /* __NR_sched_getaffinity */
+# elif KMP_ARCH_MIPS
+#  ifndef __NR_sched_setaffinity
+#   define __NR_sched_setaffinity  4239
+#  elif __NR_sched_setaffinity != 4239
+#   error Wrong code for setaffinity system call.
+#  endif /* __NR_sched_setaffinity */
+#  ifndef __NR_sched_getaffinity
+#   define __NR_sched_getaffinity  4240
+#  elif __NR_sched_getaffinity != 4240
+#   error Wrong code for getaffinity system call.
+#  endif /* __NR_sched_getaffinity */
+# elif KMP_ARCH_MIPS64
+#  ifndef __NR_sched_setaffinity
+#   define __NR_sched_setaffinity  5195
+#  elif __NR_sched_setaffinity != 5195
+#   error Wrong code for setaffinity system call.
+#  endif /* __NR_sched_setaffinity */
+#  ifndef __NR_sched_getaffinity
+#   define __NR_sched_getaffinity  5196
+#  elif __NR_sched_getaffinity != 5196
+#   error Wrong code for getaffinity system call.
+#  endif /* __NR_sched_getaffinity */
+# else
 #error Unknown or unsupported architecture
 #endif /* KMP_ARCH_* */
 #elif KMP_OS_FREEBSD
