Index: llvm-toolchain-snapshot_15~++20220724113059+7feab85df8e8/libcxx/include/atomic
===================================================================
--- llvm-toolchain-snapshot_15~++20220724113059+7feab85df8e8.orig/libcxx/include/atomic
+++ llvm-toolchain-snapshot_15~++20220724113059+7feab85df8e8/libcxx/include/atomic
@@ -2687,9 +2687,13 @@ typedef conditional<_LIBCPP_CONTENTION_L
 #endif
 
 #if !defined(_LIBCPP_NO_LOCK_FREE_TYPES)
+#ifndef NO_LOCK_FREE
 typedef atomic<__libcpp_signed_lock_free> atomic_signed_lock_free;
 typedef atomic<__libcpp_unsigned_lock_free> atomic_unsigned_lock_free;
 #endif
+#endif
+
+#undef NO_LOCK_FREE
 
 #define ATOMIC_FLAG_INIT {false}
 #define ATOMIC_VAR_INIT(__v) {__v}
