Index: llvm-toolchain-snapshot_14~++20211114120952+5b4bfd8c2415/llvm/lib/Support/Triple.cpp
===================================================================
--- llvm-toolchain-snapshot_14~++20211114120952+5b4bfd8c2415.orig/llvm/lib/Support/Triple.cpp
+++ llvm-toolchain-snapshot_14~++20211114120952+5b4bfd8c2415/llvm/lib/Support/Triple.cpp
@@ -1830,7 +1830,7 @@ StringRef Triple::getARMCPUForArch(Strin
     case llvm::Triple::EABIHF:
     case llvm::Triple::GNUEABIHF:
     case llvm::Triple::MuslEABIHF:
-      return "arm1176jzf-s";
+      return "cortex-a7";
     default:
       return "arm7tdmi";
     }
