Index: llvm-toolchain-15_15.0.1~++20220920090008+b73d2c8c720a/llvm/include/llvm/Target/TargetMachine.h
===================================================================
--- llvm-toolchain-15_15.0.1~++20220920090008+b73d2c8c720a.orig/llvm/include/llvm/Target/TargetMachine.h
+++ llvm-toolchain-15_15.0.1~++20220920090008+b73d2c8c720a/llvm/include/llvm/Target/TargetMachine.h
@@ -447,13 +447,13 @@ public:
                                      raw_pwrite_stream &, raw_pwrite_stream *,
                                      CodeGenFileType, CGPassBuilderOption,
                                      PassInstrumentationCallbacks *) {
-    return make_error<StringError>("buildCodeGenPipeline is not overriden",
+    return make_error<StringError>("buildCodeGenPipeline is not overridden",
                                    inconvertibleErrorCode());
   }
 
   virtual std::pair<StringRef, bool> getPassNameFromLegacyName(StringRef) {
     llvm_unreachable(
-        "getPassNameFromLegacyName parseMIRPipeline is not overriden");
+        "getPassNameFromLegacyName parseMIRPipeline is not overridden");
   }
 
   /// Add passes to the specified pass manager to get machine code emitted with
