Index: llvm-toolchain-snapshot_15~++20220724113059+7feab85df8e8/compiler-rt/lib/scudo/standalone/CMakeLists.txt
===================================================================
--- llvm-toolchain-snapshot_15~++20220724113059+7feab85df8e8.orig/compiler-rt/lib/scudo/standalone/CMakeLists.txt
+++ llvm-toolchain-snapshot_15~++20220724113059+7feab85df8e8/compiler-rt/lib/scudo/standalone/CMakeLists.txt
@@ -145,6 +145,13 @@ if(COMPILER_RT_DEFAULT_TARGET_ARCH MATCH
   list(APPEND SCUDO_LINK_LIBS atomic)
 endif()
 
+if(CMAKE_HOST_SYSTEM_PROCESSOR MATCHES "mips" OR
+   CMAKE_HOST_SYSTEM_PROCESSOR MATCHES "mips64" OR
+   CMAKE_HOST_SYSTEM_PROCESSOR MATCHES "mipsel" OR
+   CMAKE_HOST_SYSTEM_PROCESSOR MATCHES "mips64el")
+  list(APPEND SCUDO_LINK_LIBS atomic)
+endif()
+
 if(COMPILER_RT_HAS_SCUDO_STANDALONE)
   add_compiler_rt_object_libraries(RTScudoStandalone
     ARCHS ${SCUDO_STANDALONE_SUPPORTED_ARCH}
