Index: llvm-toolchain-snapshot_15~++20220407011631+46f0e2ceb487/clang/lib/Basic/FileManager.cpp
===================================================================
--- llvm-toolchain-snapshot_15~++20220407011631+46f0e2ceb487.orig/clang/lib/Basic/FileManager.cpp
+++ llvm-toolchain-snapshot_15~++20220407011631+46f0e2ceb487/clang/lib/Basic/FileManager.cpp
@@ -599,6 +599,12 @@ FileManager::getNoncachedStatValue(Strin
   return std::error_code();
 }
 
+// For GNU Hurd
+#if defined(__GNU__) && !defined(PATH_MAX)
+# define PATH_MAX 4096
+#endif
+
+
 void FileManager::GetUniqueIDMapping(
     SmallVectorImpl<const FileEntry *> &UIDToFiles) const {
   UIDToFiles.clear();
Index: llvm-toolchain-snapshot_15~++20220407011631+46f0e2ceb487/lldb/include/lldb/lldb-defines.h
===================================================================
--- llvm-toolchain-snapshot_15~++20220407011631+46f0e2ceb487.orig/lldb/include/lldb/lldb-defines.h
+++ llvm-toolchain-snapshot_15~++20220407011631+46f0e2ceb487/lldb/include/lldb/lldb-defines.h
@@ -15,6 +15,11 @@
 #define INT32_MAX 2147483647
 #endif
 
+// For GNU Hurd
+#if defined(__GNU__) && !defined(PATH_MAX)
+# define PATH_MAX 4096
+#endif
+
 #if !defined(UINT32_MAX)
 #define UINT32_MAX 4294967295U
 #endif
Index: llvm-toolchain-snapshot_15~++20220407011631+46f0e2ceb487/polly/lib/External/ppcg/cuda_common.c
===================================================================
--- llvm-toolchain-snapshot_15~++20220407011631+46f0e2ceb487.orig/polly/lib/External/ppcg/cuda_common.c
+++ llvm-toolchain-snapshot_15~++20220407011631+46f0e2ceb487/polly/lib/External/ppcg/cuda_common.c
@@ -15,6 +15,11 @@
 #include "cuda_common.h"
 #include "ppcg.h"
 
+// For GNU Hurd
+#if defined(__GNU__) && !defined(PATH_MAX)
+# define PATH_MAX 4096
+#endif
+
 /* Open the host .cu file and the kernel .hu and .cu files for writing.
  * Add the necessary includes.
  */
