Index: llvm-toolchain-snapshot_10~+20191020075550+5b8546023f3/lldb/source/Host/freebsd/Host.cpp
===================================================================
--- llvm-toolchain-snapshot_10~+20191020075550+5b8546023f3.orig/lldb/source/Host/freebsd/Host.cpp
+++ llvm-toolchain-snapshot_10~+20191020075550+5b8546023f3/lldb/source/Host/freebsd/Host.cpp
@@ -10,6 +10,13 @@
 #include <sys/types.h>
 
 #include <sys/exec.h>
+#ifdef __FreeBSD_kernel__
+#  ifdef __amd64__
+#  include </usr/include/x86_64-kfreebsd-gnu/sys/kglue/sys/types.h>
+#  else
+#  include </usr/include/i386-kfreebsd-gnu/sys/kglue/sys/types.h>
+#  endif
+#endif
 #include <sys/proc.h>
 #include <sys/ptrace.h>
 #include <sys/sysctl.h>
